site/src/pages/WorkspacePage/WorkspaceNotifications.tsx

```diff
@@ -0,0 +1,343 @@
+import { workspaceResolveAutostart } from "api/queries/workspaceQuota";
+import { Template, TemplateVersion, Workspace } from "api/typesGenerated";
+import { AlertProps } from "components/Alert/Alert";
+import { FC, ReactNode, useEffect, useState } from "react";
+import { useQuery } from "react-query";
+import { WorkspacePermissions } from "./permissions";
+import dayjs from "dayjs";
+import { useIsWorkspaceActionsEnabled } from "components/Dashboard/DashboardProvider";
+import formatDistanceToNow from "date-fns/formatDistanceToNow";
+import { Pill } from "components/Pill/Pill";
+import InfoOutlined from "@mui/icons-material/InfoOutlined";
+import {
+  Popover,
+  PopoverContent,
+  PopoverTrigger,
+} from "components/Popover/Popover";
+import { Interpolation, Theme, useTheme } from "@emotion/react";
+import Button, { ButtonProps } from "@mui/material/Button";
+import { ThemeRole } from "theme/experimental";
+import WarningRounded from "@mui/icons-material/WarningRounded";
+
+type Notification = {
+  title: string;
+  severity: AlertProps["severity"];
+  detail?: ReactNode;
+  actions?: ReactNode;
+};
+
+type WorkspaceNotificationsProps = {
+  workspace: Workspace;
+  template: Template;
+  permissions: WorkspacePermissions;
+  onRestartWorkspace: () => void;
+  onUpdateWorkspace: () => void;
+  onActivateWorkspace: () => void;
+  latestVersion?: TemplateVersion;
+};
+
+export const WorkspaceNotifications: FC<WorkspaceNotificationsProps> = (
+  props,
+) => {
+  const {
+    workspace,
+    template,
+    latestVersion,
+    permissions,
+    onRestartWorkspace,
+    onUpdateWorkspace,
+    onActivateWorkspace,
+  } = props;
+  const notifications: Notification[] = [];
+
+  // Outdated
+  const canAutostartResponse = useQuery(
+    workspaceResolveAutostart(workspace.id),
+  );
+  const canAutostart = !canAutostartResponse.data?.parameter_mismatch ?? false;
```

Comment:
It feels like there are some missing parentheses, just because of the `?? false` part:

- Fetch hasn't completed yet – `parameter_mismatch` is `undefined`, and the negation turns it into `true`. The `?? false` never triggers
- Fetch has completed and there is no mismatch – `parameter_mismatch` becomes `false`, and is negated into `true`. The `?? false` never triggers
- Fetch has completed and there is a mismatch – `parameter_mismatch` becomes `true`, and is negated into `false`. The `?? false` never triggers

My main worry is that this would be `true` during loading/error states. I'm guessing this would need to be something like:
```ts
const canAutostart = !(canAutostartResponse.data?.parameter_mismatch ?? true);
```

Reply:
my main worry is that this one line is trying to do much. 😅 if we're already struggling to understand what this does, it should probably be broken apart

Reply:
Better?
```tsx
const isParameterMismatch =
    canAutostartQuery.data?.parameter_mismatch ?? false;
  const canAutostart = !isParameterMismatch;
```

Reply:
Yeah, I think so